### PR TITLE
feature/article-parser

### PR DESCRIPTION
**Pull request by Olga232**

### CODE REVIEW

File: src/main/java/baza/trainee/service/impl/ArticleServiceImpl.java [hunk 1]
```diff
@@ -1,28 +1,43 @@
...
+import java.util.List;
 
 @Service
 @RequiredArgsConstructor
 public class ArticleServiceImpl implements ArticleService {
 
     private final ArticleRepository articleRepository;
-    private final ArticleMapper articleMapper;
+    private final ObjectMapper objectMapper;
+    @Value("${static.json.articles}")
```
Comment (malyshevhen) [inline src/main/java/baza/trainee/service/impl/ArticleServiceImpl.java:27]: Add new line before annotation.

Comment (Olga232) [reply src/main/java/baza/trainee/service/impl/ArticleServiceImpl.java:27]: Added

**Review by malyshevhen — APPROVED** [review]

File: src/main/java/baza/trainee/service/impl/ArticleServiceImpl.java [hunk 2]
```diff
@@ -1,28 +1,44 @@
...
+        try {
+            final var inputStream = new ClassPathResource(resource).getInputStream();
+            final var fileData = FileCopyUtils.copyToByteArray(inputStream);
+            final CollectionType collectionType =
+                    objectMapper.getTypeFactory().constructCollectionType(List.class, Article.class);
+            final List<Article> articleList = objectMapper.readValue(fileData, collectionType);
+            articleList.forEach(articleRepository::save);
+        } catch (IOException e) {
+            throw new RuntimeException(e.getMessage());
+        }
```
Comment (malyshevhen) [inline src/main/java/baza/trainee/service/impl/ArticleServiceImpl.java]:
```suggestion
        try(var inputStream = new ClassPathResource(resource).getInputStream()) {
            final var fileData = FileCopyUtils.copyToByteArray(inputStream);
            final CollectionType collectionType =
                    objectMapper.getTypeFactory().constructCollectionType(List.class, Article.class);
            final List<Article> articleList = objectMapper.readValue(fileData, collectionType);
            articleList.forEach(articleRepository::save);
        } catch (IOException e) {
            throw new RuntimeException(e.getMessage());
        }
```

File: src/main/java/baza/trainee/service/impl/ArticleServiceImpl.java [hunk 3]
```diff
@@ -1,28 +1,44 @@
...
+    @Transactional
+    public void saveStaticArticles() {
+        try {
+            final var inputStream = new ClassPathResource(resource).getInputStream();
+            final var fileData = FileCopyUtils.copyToByteArray(inputStream);
+            final CollectionType collectionType =
+                    objectMapper.getTypeFactory().constructCollectionType(List.class, Article.class);
+            final List<Article> articleList = objectMapper.readValue(fileData, collectionType);
+            articleList.forEach(articleRepository::save);
+        } catch (IOException e) {
```
Comment (malyshevhen) [inline src/main/java/baza/trainee/service/impl/ArticleServiceImpl.java:40]: Add StreamReadException and DatabindException to catch block

File: src/main/java/baza/trainee/service/impl/ArticleServiceImpl.java [hunk 4]
```diff
@@ -1,28 +1,44 @@
...
-                .orElseThrow(getNotFoundExceptionSupplier(Article.class ,"title: " + title));
+    @PostConstruct
+    @Transactional
+    public void saveStaticArticles() {
+        try {
+            final var inputStream = new ClassPathResource(resource).getInputStream();
+            final var fileData = FileCopyUtils.copyToByteArray(inputStream);
+            final CollectionType collectionType =
+                    objectMapper.getTypeFactory().constructCollectionType(List.class, Article.class);
+            final List<Article> articleList = objectMapper.readValue(fileData, collectionType);
```
Comment (malyshevhen) [inline src/main/java/baza/trainee/service/impl/ArticleServiceImpl.java]: Use var or explicit type.

**Review by Anatolii-Omelchenko — APPROVED** [review]

**Review by malyshevhen — APPROVED** [review]

**Pull request closed without merging**